src/main/scala/com/databricks/labs/overwatch/env/Workspace.scala

```diff
@@ -98,10 +98,19 @@ class Workspace(config: Config) extends SparkSessionWrapper {
    */
   def getPoolsDF: DataFrame = {
     val poolsEndpoint = "instance-pools/list"
-    ApiCallV2(config.apiEnv,poolsEndpoint)
-      .execute()
-      .asDF()
-      .withColumn("organization_id", lit(config.organizationId))
+    try {
+      ApiCallV2(config.apiEnv, poolsEndpoint)
+        .execute()
+        .asDF()
+        .withColumn("organization_id", lit(config.organizationId))
+    }
+    catch {
+      case e: Exception =>
```

Comment:
Put try catch only on  .asDF() method,
 If we put try-catch on .execute() then we might miss other exceptions like no access/invalid URL etc.

Reply:
changed the type of exception catching to ApiCallEmptyResponse. This should rule out empty responses from API and raise exceptions in case of any other exception. Please check.

Reply:
agree with @Sriram-databricks -- the right place to put this error handler is in the APICallV2 asDF function to ensure that the errors bubble up to the pipReport. 

If I understand correctly, the actual error is not the API call as that will work fine and return 0 results, but the issue is rather that when we try to convert it to a DF and select the nested column, the col doesn't exist and is causing the breakage. I think we need a generic handler for this in asDF in APICallV2 so that any APICall that is tried to be converted to a DF but is missing the expected column should error.